src/coreclr/tools/Common/Compiler/Logging/MessageContainer.cs

```diff
@@ -87,6 +103,23 @@ internal static MessageContainer CreateErrorMessage(string text, int code, strin
             return CreateWarningMessageContainer(context, text, code, origin, subcategory);
         }
 
+        /// <summary>
+        /// Create a warning message.
+        /// </summary>
+        /// <param name="context">Context with the relevant warning suppression info.</param>
+        /// <param name="origin">Filename or member where the warning is coming from</param>
+        /// <param name="id">Unique warning ID. Please see https://github.com/dotnet/linker/blob/main/docs/error-codes.md
+        /// for the list of warnings and possibly add a new one</param>
+        /// <param name="args">Additional arguments to form a humanly readable message describing the warning</param>
+        /// <returns>New MessageContainer of 'Warning' category</returns>
+        internal static MessageContainer? CreateWarningMessage(Logger context, MessageOrigin origin, DiagnosticId id, params string[] args)
+        {
+            //if (!((int)id > 2000 && (int)id <= 6000))
```

Comment:
Why is this commented out? Is this a sync from the illinker repo? If so, could you update src/coreclr/tools/Common/Compiler/Logging/ReferenceSource?

Reply:
> Why is this commented out

I copy the method from the one currently in the MessageContainer.cs and has commented that block of code, so I commented out.

I didn't notice that reference source, I will update.

Reply:
Now that there are failures seems to be because we try to log a warning for IL1005 which is an error code. It falls out of the range and fails some tests.